service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/database/AzureDatabaseUtilsRunner.java

```diff
@@ -361,9 +373,39 @@ public void testDatabaseConnect(
   private void runAzureDatabaseUtils(
       AzureCloudContext azureCloudContext, UUID workspaceId, V1Pod podDefinition, String namespace)
       throws InterruptedException {
+    runAzureDatabaseUtils(
+        azureCloudContext, workspaceId, podDefinition, new HashMap<>(), namespace);
+  }
+
+  private void runAzureDatabaseUtils(
+      AzureCloudContext azureCloudContext,
+      UUID workspaceId,
+      V1Pod podDefinition,
+      Map<String, String> secretStringData,
+      String namespace)
+      throws InterruptedException {
     var aksApi = kubernetesClientProvider.createCoreApiClient(azureCloudContext, workspaceId);
+
     // strip underscores to avoid violating azure's naming conventions for pods
     var safePodName = podDefinition.getMetadata().getName();
+    var doSecretCleanup = false;
+    if (getPodDefinitionEnvVarsWithSecretRefs(podDefinition).size() > 0
+        && secretStringData.isEmpty()) {
+      throw new RuntimeException(
+          "definition of pod {} contains env vars that refer to secrets, but no secret data was provided."
+              .formatted(safePodName));
+    } else if (secretStringData.size() > 0) {
+      try {
+        createSecret(
+            aksApi,
+            secretStringData,
+            safePodName, // give the secret the same name as the pod
+            namespace);
+      } catch (ApiException e) {
+        throw new RetryException(e);
+      }
+    }
```

Comment:
I think it would make this code more readable if `secretStringData.size() > 0` check is inside `createSecret` along with the exception handling. And do a similar check inside `deleteSecret`